packages/web-app/src/pages/proposal.tsx

```diff
@@ -76,6 +86,55 @@ const Proposal: React.FC = () => {
     }
   }, [editor, proposal]);
 
+  const findDecoder = useCallback(
+    (action: DaoAction) => {
+      const functionParams = client?.decoding.findInterface(action.data);
+      switch (functionParams?.functionName) {
+        case 'withdraw':
+          return decodeWithdrawToAction(
+            action.data,
+            client,
+            apolloClient,
+            network
+          );
+        default:
+          return {} as ActionWithdraw;
+      }
+    },
+    [apolloClient, client, network]
```

Comment:
If you're return a promise in the other case it's gonna make things easier if you return a Promise here too. You can do this by wrapping the returned object with the `reolve` function

```suggestion
          return Promise.resolve({} as ActionWithdraw);
```